mindsdb/integrations/handlers/discord_handler/discord_handler.py

```diff
@@ -0,0 +1,185 @@
+    HandlerResponse as Response,
+    RESPONSE_TYPE,
+)
+
+discord_bot = None
+
+class DiscordHandler(APIHandler):
+    """
+    The Discord handler implementation.
+    """
+
+    name = 'discord'
+
+    def __init__(self, name: str, **kwargs):
+        """
+        Initialize the handler.
+        Args:
+            name (str): name of particular handler instance
+            **kwargs: arbitrary keyword arguments.
+        """
+        super().__init__(name)
+
+        connection_data = kwargs.get("connection_data", {})
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+
+        self.is_connected = False
+
+        messages = DiscordTable(self)
+        self._register_table('messages', messages)
+
+    def connect(self):
+        """
+        Set up the connection required by the handler.
+        Returns
+        -------
+        StatusResponse
+            connection object
+        """
+
+        if self.is_connected:
+            return StatusResponse(True)
+
+        url = f'https://discord.com/api/v10/applications/@me'
+        result = requests.get(
+            url,
+            headers={
+                'Authorization': f'Bot {self.connection_data["token"]}',
+                'Content-Type': 'application/json',
+            },
+        )
+
+        if result.status_code != 200:
+            raise ValueError(result.text)
+
+        self.is_connected = True
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Check connection to the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        response = StatusResponse(False)
+
+        try:
+            self.connect()
+            response.success = True
+        except Exception as e:
+            response.error_message = e
+            log.logger.error(f'Error connecting to Discord: {response.error_message}')
+
+        self.is_connected = response.success
+
+        return response
+
+    def native_query(self, query: str = None) -> StatusResponse:
+        """Receive and process a raw query.
+        Parameters
+        ----------
+        query : str
+            query in a native format
+        Returns
+        -------
+        StatusResponse
+            Request status
+        """
+        method_name, params = FuncParser().from_string(query)
+
+        df = self.call_discord_api(method_name, params)
+
+        return Response(RESPONSE_TYPE.TABLE, data_frame=df)
+    
+    def utc_to_snowflake(self, utc_date: str) -> int:
+        """
+        Convert a UTC date to a Snowflake date.
+        Args:
+            utc_date (str): the UTC date
+        Returns:
+            int
+        """
+        # https://discord.com/developers/docs/reference#snowflakes
+        return str(
+            int(parse_utc_date(utc_date).timestamp() * 1000 - 1420070400000) << 22
+        )
+
+    def call_discord_api(
+            self, method_name: str, params: dict = None, filters: list = None
```

Comment:
Can we change the `method_name` parameter to something like `operation`? method_name makes is seem slightly 'hacky' somehow.